mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -1480,17 +1480,24 @@ struct DropUnitDimFromElementwiseOps final
   using OpTraitRewritePattern::OpTraitRewritePattern;
   LogicalResult matchAndRewrite(Operation *op,
                                 PatternRewriter &rewriter) const override {
-    if (op->getNumResults() != 1)
+    if (op->getNumResults() != 1 || op->getNumRegions() != 0)
```

Comment:
What cases does this condition catch: ` op->getNumRegions() != 0)`?

Reply:
I was inspired by the pattern `ReorderElementwiseOpsOnTranspose`, which does this check. I don't know if there is an elementwise op with region now, but I think it is a little more future-proof, as currently this pattern doesn't handle the op with region.

Reply:
I skipped this check because I couldn't find an example for which it would be required :) But I also couldn't find anything that would guarantee that "elementwise" will have no regions 🤷🏻. Anyway, this is fine and like you said - makes things future-proof.